### PR TITLE
[CARBONDATA-3659] Fix issues with alluxio without host and port.

### PR DESCRIPTION
**Pull request by ravipesala**
### Why is this PR needed?
 When alluxio path is provided without host and port like alluxio:///user/warehouse then carbon cannot read or write data because of path comparison fails and extracting parent path fails.
 
 ### What changes were proposed in this PR?
  Use Path object to compare paths. And use string utils to extract the parent path.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1565/

File: core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java [hunk 1]
```diff
@@ -554,6 +555,21 @@ public short getDefaultReplication() {
     return fileSystem.getDefaultReplication(path);
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java]: add `{` and `}`

Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java]: ok

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1585/

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java [hunk 2]
```diff
@@ -137,12 +137,14 @@ public void init(DataMapModel dataMapModel)
         // if the segment data is written in tablepath then no need to store whole path of file.
         !blockletDataMapInfo.getFilePath().startsWith(
             blockletDataMapInfo.getCarbonTable().getTablePath())) {
-      filePath = path.getParent().toString().getBytes(CarbonCommonConstants.DEFAULT_CHARSET);
+      filePath =
+          path.substring(0, path.lastIndexOf("/")).getBytes(CarbonCommonConstants.DEFAULT_CHARSET);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java]: I found that you can use `FilenameUtils.getFullPathNoEndSeparator(file)` from Apache Commons

File: core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java [hunk 3]
```diff
@@ -189,7 +189,7 @@ private static BlockMetaInfo createBlockMetaInfo(
         CarbonFile carbonFile = FileFactory.getCarbonFile(carbonDataFile);
         return new BlockMetaInfo(new String[] { "localhost" }, carbonFile.getSize());
       default:
-        return fileNameToMetaInfoMapping.get(carbonDataFile);
+        return fileNameToMetaInfoMapping.get(FileFactory.getCarbonFile(carbonDataFile).getPath());
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java]: please add comment why this is required? Can we add a case for Alluxio separately?

File: core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java [hunk 4]
```diff
@@ -198,9 +198,10 @@ private static BlockMetaInfo createBlockMetaInfo(
     Set<TableBlockIndexUniqueIdentifier> tableBlockIndexUniqueIdentifiers = new HashSet<>();
     Map<String, String> indexFiles = segment.getCommittedIndexFile();
     for (Map.Entry<String, String> indexFileEntry : indexFiles.entrySet()) {
-      Path indexFile = new Path(indexFileEntry.getKey());
+      String indexFile = indexFileEntry.getKey();
       tableBlockIndexUniqueIdentifiers.add(
-          new TableBlockIndexUniqueIdentifier(indexFile.getParent().toString(), indexFile.getName(),
+          new TableBlockIndexUniqueIdentifier(indexFile.substring(0, indexFile.lastIndexOf("/")),
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java]: use FileNameUtils from Apache Commons

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java (hunk 2 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java]: ok

File: core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java (hunk 3 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java]: ok

File: core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java (hunk 4 above)
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java]: ok

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1637/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**